### PR TITLE
Issue #206 -ec2 check to support IMDSv1 and IMDSv2 + moving env vars to conf file

### PR DESCRIPTION
**Pull request by orensharoni**
we added support of IMDSv2 to the EC2 check with a fall back to IMDSv1,
also, some nginx env declarations were moved from the environment file to the nginx.conf file

### CODE REVIEW

**Comment by 4141done** [comment]
Thank you for your contribution, @orensharoni  - have been able to verify that this fixes your issue from https://github.com/nginxinc/nginx-s3-gateway/issues/206 ?

I'll stand up an ec2 env and give it a check too.

**Comment by 4141done** [comment]
> yes, I used it on my dev setup and the ec2 test and deployment work, the access_key and access_token should be also verified if posible

Excellent, thank you for confirming.  I'll do a quick check on my end and then merge

File: standalone_ubuntu_oss_install.sh [hunk 1]
```diff
@@ -312,12 +316,12 @@ EOF
 # Only include these env vars if we are not using a instance profile credential
 # to obtain S3 permissions.
 if [ $uses_iam_creds -eq 0 ]; then
-  cat >> "/etc/nginx/environment" << EOF
+  cat >> "/etc/nginx/nginix.conf" << EOF
```
Comment (4141done) [inline standalone_ubuntu_oss_install.sh:319]:
```suggestion
  cat >> "/etc/nginx/nginx.conf" << EOF
```

File: standalone_ubuntu_oss_install.sh [hunk 2]
```diff
@@ -312,12 +316,12 @@ EOF
...
 env AWS_ACCESS_KEY_ID;
 env AWS_SECRET_ACCESS_KEY;
 EOF
   if [[ -v AWS_SESSION_TOKEN ]]; then
-    cat >> "/etc/nginx/environment" << EOF
+    cat >> "/etc/nginx/nginix.conf" << EOF
```
Comment (4141done) [inline standalone_ubuntu_oss_install.sh:324]:
```suggestion
    cat >> "/etc/nginx/nginx.conf" << EOF
```

**Review by 4141done — CHANGES_REQUESTED** [review]
Thanks again for your contribution.  I tested these changes with the token approach and was able to verify it works well.

If you can correct the two typoes I highlighted we can go ahead and merge.

File: standalone_ubuntu_oss_install.sh (hunk 1 above)
Comment (orensharoni) [reply standalone_ubuntu_oss_install.sh]: done

File: standalone_ubuntu_oss_install.sh (hunk 2 above)
Comment (orensharoni) [reply standalone_ubuntu_oss_install.sh]: done

**Review by 4141done — APPROVED** [review]
Thank you again for your contribution! Also special thanks to @HighOnMikey
for noticing the wrong file path in the env var inclusion portion

**Pull request merged**